compiler/ballerina-lang/src/main/java/io/ballerina/projects/bala/BalaProject.java

```diff
@@ -126,4 +126,9 @@ private boolean isFilePathInProject(Path filepath) {
         }
         return true;
     }
+
+    @Override
+    public Path targetDir() {
+        throw new UnsupportedOperationException("Target Dir is not supported for BalaProject");
```

Comment:
Shall we reword this to `target directory`

Reply:
Changed to "Target directory is not supported for Balaproject"